experimental/emulated_vault/opt/emulated-vault/vault-debug

```diff
@@ -0,0 +1,186 @@
+                                  action="store_true", dest="setParameter", default=False, 
+                                  help="Set parameter. Parameter relative path should be provided.")
+
+    operationModeGroup.add_option("--delete-parameter",
+                                  action="store_true", dest="deleteParameter", default=False, 
+                                  help="Delete parameter. Parameter relative path should be provided.")
+
+    operationModeGroup.add_option("--search-parameters",
+                                  action="store_true", dest="searchParameters", default=False, 
+                                  help="Search parameter. Parameter relative path should be provided.")
+
+    (options, args) = parser.parse_args()
+
+    global adapter
+    adapterType = generalCfg.get("adapter-type")
+    if not adapterType:
+        logger.error("Missing adapter type cfg")
+        sys.exit(1)
+    elif adapterType == "fs":
+        adapter = adapter.fs.Fs(logger=logger)
+    else:
+        logger.error("Invalid adapter type '%s'", adapterType)
+        sys.exit(1)
+
+    if not adapter.init_cfg(config):
+        logger.error("Failed adapter initialization")
+        sys.exit(1)        
+    if not adapter.init_ping():
+        logger.error("Failed adaper ping initialization")
+        sys.exit(1)        
+
+    if options.ping:
+        if len(args) != 0:
+            parser.error("Command should get no arguments.")
+        success, value = adapter.ping()
+        if not success:
+            sys.exit(1)
+        print(value)
+        sys.exit(0)
+
+    if options.getParameter:
+        if len(args) != 1:
+            parser.error("Command should get a single argument - parameter key.")
+        success, value = adapter.getParameter(args[0])
+        if not success:
+            sys.exit(1)
+        if value is None:
+            sys.exit(1)
+        print(value)
+        sys.exit(0)
+
+    if options.searchParameters:
+        if len(args) != 1:
+            parser.error("Command should get a single argument - parameter key.")
+        success, value = adapter.searchParameters(args[0], keyFilters={}, filters={})
+        if not success:
+            sys.exit(1)
+        print(json.dumps(value))
+        sys.exit(0)
+
+    if options.setParameter:
+        if len(args) != 2:
+            parser.error("Command should get 2 arguments - parameter key and value.")
+        success = adapter.setParameter(args[0], args[1])
+        if not success:
+            sys.exit(1)
+        sys.exit(0)
+
+    if options.deleteParameter:
+        if len(args) != 1:
+            parser.error("Command should get a single argument - parameter key.")
+        success = adapter.deleteParameter(args[0])
+        if not success:
+            sys.exit(1)
+        sys.exit(0)
+
+    logger.error("Operation is not set")
+    parser.print_help()
+    parser.error("No operation mode specified")
+    sys.exit(1)
+
+
+
```

Comment:
current score of this file is -4.72/10. There's no established baseline for what experimental code has to score - and I'm not sure there should be - but you can increase it to a 5.92 by making two small changes:

- use tabs for indentation
- The naming conventions want all global scope variables to be UPPER_CASE, so currently it has a problem with every declaration under your `if __name__ == '__main__':` statement. There's two ways to fix that, one is to make them all UPPER_CASE, but that's probably not what you want. The other is to wrap it in a function like:

```python
def main():
	'''The main logic'''
	# other code here...

if __name__ == '__main__':
	main()
```
Personally I like to do this and make `main` return an integer return code so that it can be called from an interactive prompt without exiting the session. Then under `if __name__ ==  'main':` I just do `sys.exit(main())`. That's not a requirement or anything, just my hot tip of the day.